docs/contributor/performance-tests/metrics/metrics.md

```diff
@@ -0,0 +1,690 @@
+        prometheus.io/port: "8080"
+        prometheus.io/scrape: "true"
+    spec:
+      containers:
+      - name: avalanche-metric-load-generator
+        image: quay.io/freshtracks.io/avalanche
+        imagePullPolicy: IfNotPresent
+        args:
+        - --metric-count=500
+        - --series-count=20
+        - --port=8080
+        resources:
+          limits:
+            memory: "128Mi"
+            cpu: "200m"
+        ports:
+        - containerPort: 8080
+          protocol: TCP
+          name: metrics
+```
+
+Service for metric agent metrics
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    prometheus.io/port: "8888"
+    prometheus.io/scrape: "true"
+    prometheus.io/scheme: "http"
+  labels:
+    app.kubernetes.io/name: telemetry-metric-agent
+  name: telemetry-metric-agent-metrics
+  namespace: kyma-system
+spec:
+  clusterIP: None
+  ports:
+  - name: http-metrics
+    port: 8888
+    protocol: TCP
+    targetPort: 8888
+  selector:
+    app.kubernetes.io/name: telemetry-metric-agent
+  sessionAffinity: None
+  type: ClusterIP
+```
+
+## Testcases
+
+### Assumptions
+
+We tweak metrics and series value with 10 labels in each metric data point. The test is executed for 1 hour to have a stabilized output. Additionally, the load is scaled slowly to prevent spikes which might cause OOMs of metric agent.
+
+We identified following test cases:
+1. Multiple pods all running on a single node and export metrics (to find how many workloads supported)
+2. Workload generating huge amount of metrics (To understand how scraping works when the workload exposes several MB of metrics)
+3. Have multiple workloads across different nodes (To understand prometheus SDS behaviour with multiple services)
+4. Verify sample limit test
+5. Multiple pods and services scraped from multiple receivers
+
+### Multiple pods all running on a single node and export metrics
+Setup:
+
+Avalanche load generator configuration per pod:
+- Metric count: 500
+- Metric series: 20
+- Number of labels: 10
+
+To simulate a smooth ramp-up and avoid huge data flood at the beginning, load generator was started with 5 instances, instance count was increased at 5 instances every minute until peak was reached.
+
+This test was executed on a single node installation to determine the limits of a single metric agent, after the load generator reached `28` instances, metric agent hits the limits and stays stable with this setup. Any load above this setup would cause memory limiter to refuse incoming traffic.
+
+The graph below shows the average metric points processed by the `prometheus metric receiver` per second, agent reaches an average of `14K metric points/sec`
+![Peak accepted metric points](./assets/overall-peak-metric-point.jpg)
+
+Following graph shows the refused metric points from the configured `memory_limiter`, with traffic above 14K metric points/sec results in  metrics being refused by the `memory_limiter`. 
```

Comment:
```suggestion
The following graph shows the refused metric points from the configured `memory_limiter`, with traffic above 14K metric points/sec results in  metrics being refused by the `memory_limiter`. 
```
Same here, why are some elements formatted in code font? Does any of our [guidelines](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/03-formatting.md#use-code-font-for-these-items) apply?

Reply:
Reworked